tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/OptimizerCore.scala

```diff
@@ -1554,6 +1591,62 @@ private[optimizer] abstract class OptimizerCore(
       case LongToOctalStr =>
         contTree(Apply(firstArgAsRTLong, LongImpl.toOctalString, Nil)(StringClassType))
 
+      // scala.collection.mutable.ArrayBuilder
+
+      case GenericArrayBuilderResult =>
+        val List(runtimeClass, array) = newArgs
+        val (resultType, isExact) = runtimeClass match {
+          case ClassOf(elemType) => (ArrayType(elemType), true)
+          case _                 => (AnyType, false)
+        }
+        cont(PreTransTree(CallHelper("makeNativeArrayWrapper",
+            CallHelper("arrayDataOf",
+                CallHelper("classDataOf", runtimeClass)(AnyType))(AnyType),
```

Comment:
I don't like this since it exposes the existence of class data to the optimizer. Could we make a single dummy that takes a `j.l.Class` and a `js.Array` and produces the wrapped array?